### ISSUE TITLE
find query returning wrong results with IndexedDB RxStorage adapter

### ISSUE THREAD

**Issue by mauricehaider** [pubkey/rxdb#4831]
Following situation:

I have multiple documents in a collection, e.g

```
[{
"name": "Project 1",
"type: "project"
},
{
"name": "Project 2",
"type: "project"
},
{
"name": "Project Share",
"type: "project-sharing"
}]

```
When trying to find all documents with 'type' = 'project' we would use the following query:

```
this.dbItems
                .find({
                    selector: {
                        type: {
                            $eq: "project",
                        },
                    },
                })
                .exec()
```

The query in this case would not only return the 2 project documents, it also returns the third document with a type of "project-sharing". I guess the query does not compare for an actual equality but returns all documents **that start with** the searched selector string. In my case the type "project-sharing" also starts with "project", but as we use the "$eq" comparator **it should only return the exact match** "project".

This wrong query result only happens when using the premium IndexedDB RxStorage adapter ....
```
this.rxdb = await createRxDatabase({
                name: 'dp-db', // the name of the database
                storage: getRxStorageIndexedDB({
                    batchSize: 300,
                }),
            });
```
It doesn't occur when using e.g dexie

**Comment by pubkey** [comment]
@mauricehaider Are you on the latest rxdb version? There was a similar problem which was fixed in the latest release.

**Comment by mauricehaider** [comment]
> @mauricehaider Are you on the latest rxdb version? There was a similar problem which was fixed in the latest release.

yes same issue as #4755 , was actually fixed in 14.15.0
